doc/refactoring_design.md

```diff
@@ -0,0 +1,481 @@
+# Keeper Contract Design for Tethys
+
+***DISCLAIMER: THIS IS A WORK IN PROGRESS***
+
+This document provides technical details and describes the high-level design of keeper contract in ocean protocol.
+
+## 1. Overview
+
+### 1.1 Core Modules
+
+From the top-level point of view, there are total 9 moduels as shown in the below:
+
+<img src="img/refactoring/modules.jpg" width="1000" />
+
+* **Ocean Token**: the native ERC20 token in Ocean network
+* **Exchange**: Uniswap-like exchange built inside Ocean, where user can exchange between Ether and Ocean tokens.
+* **Service Agreement**: every service is modeled as a *service agreement*. Please see [SLA.md](https://github.com/oceanprotocol/dev-ocean/blob/feature/SLA-specs/doc/sla.md) and [smart contract prototype](https://github.com/oceanprotocol/keeper-contracts/blob/develop/contracts/SLA/ServiceAgreement.sol) for details.
```

Comment:
- `every service is modeled as a *service agreement*` change into `A service agreement is defined by a set of service conditions describing the SLA template`.
- `[smart contract prototype]` It would good to replace it into `Service Agreement implementation in Trilobite`.